infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/EphemeralWorkspaceAdapter.java

```diff
@@ -0,0 +1,152 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.namespace.pvc;
+
+import static org.eclipse.che.api.workspace.shared.Constants.MOUNT_SOURCES_ATTRIBUTE;
+import static org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesObjectUtil.newVolumeMount;
+import static org.eclipse.che.workspace.infrastructure.kubernetes.provision.LogsVolumeMachineProvisioner.LOGS_VOLUME_NAME;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodSpec;
+import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.core.model.workspace.Workspace;
+import org.eclipse.che.api.core.model.workspace.config.Volume;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.WorkspaceManager;
+import org.eclipse.che.api.workspace.server.model.impl.WorkspaceImpl;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.api.workspace.server.spi.InternalInfrastructureException;
+import org.eclipse.che.workspace.infrastructure.kubernetes.Names;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment;
+
+/**
+ * Allows to create ephemeral workspaces (with no PVC attached) based on workspace config
+ * `mountSources` attribute. If `mountSources` attribute is set to false, workspace volumes would be
+ * created as `emptyDir` regardless of the PVC strategy. When a workspace Pod is removed for any
+ * reason, the data in the `emptyDir` volume is deleted forever.
+ *
+ * @see <a href="https://kubernetes.io/docs/concepts/storage/volumes/#emptydir">emptyDir</a>
+ * @author Ilya Buziuk
+ */
+@Singleton
+public class EphemeralWorkspaceAdapter {
+  private final String EPHEMERAL_VOLUME_NAME_PREFIX = "ephemeral-che-workspace-";
+  private WorkspaceManager workspaceManager;
+
+  @Inject
+  public EphemeralWorkspaceAdapter(WorkspaceManager workspaceManager) {
+    this.workspaceManager = workspaceManager;
+  }
+
+  /**
+   * @param workspaceId
+   * @return true if workspace config contains `mountSources` attribute which is set to false. In
+   *     this case regardless of the PVC strategy, workspace volumes would be created as `emptyDir`.
+   *     When a workspace Pod is removed for any reason, the data in the `emptyDir` volume is
+   *     deleted forever
+   * @throws InternalInfrastructureException
+   */
+  public boolean isEphemeral(String workspaceId) throws InternalInfrastructureException {
+    try {
+      WorkspaceImpl workspace = workspaceManager.getWorkspace(workspaceId);
```

Comment:
We can actually avoid using WorkspaceMnaager here since we can pass WorkspaceConfig attributes here. And we can get them from Workspace object or K8s environment object. I'm not sure we really need that but would want to hear about that from @sleshchenko @ibuziuk . This doesn't block the PR.

Reply:
when I tested PR on minishift for some reason K8s environment does not seem to contain WorkspaceConfig attributes, so I had to use  WorkspaceManager instead

Reply:
@ibuziuk I think it should be considered it as a bug because it must be there https://github.com/eclipse/che/blob/27bed82d5c57dc0c95093a1cea6043cfa47877f1/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java#L740
Otherwise, WorkspaceNext won't be working on Kubernetes infra. And three weeks ago I tested it and it worked.

Reply:
@sleshchenko I will re-test and provide a comment about it. AFAIK `k8sEnv.getAttributes()` always returned emptyMap for me
`

Reply:
ok, seems I was wrong and all the config attributes are actually located in `k8sEnv.getAttributes()`
I will remove the usage of `WorkspaceMnaager`. Thanks for noticing

Reply:
done, `WorkspaceMnaager` usage has been removed from PR